### PR TITLE
Anthropic support

### PR DESCRIPTION
**Pull request by dmugtasimov**
Implementation of https://github.com/thenewboston-developers/thenewboston-Backend/issues/221

Both Anthropic and OpenAI are supported on LLMClient level. Other prompts can be switched to Anthropic just by changing prompt parameters in PromptLayer (did not test, but should work)

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -7,7 +7,10 @@
 
 # Initial Project Setup
 
-1. Install Poetry
+1. Install Python version 3.10.13 and make sure it is being used the next step and later during development
```
Comment (dmugtasimov) [inline README.md:10]: Other 3.10.x versions are also fine. The exact version is added for consistency

File: thenewboston/contributions/models/contribution.py [hunk 2]
```diff
@@ -69,11 +69,11 @@ def assess(self, save=True):
                 assessment_points = pull.assessment_points
                 assessment_explanation = pull.assessment_explanation
             case ContributionType.MANUAL.value:
-                result = OpenAIClient.get_instance().get_chat_completion(
-                    settings.GITHUB_MANUAL_CONTRIBUTION_ASSESSMENT_PROMPT_NAME,
+                result = LLMClient.get_instance().get_chat_completion(
                     input_variables={'description': self.description},
                     tracked_user=self.user,
                     tags=['manual_contribution_assessment'],
+                    **make_prompt_kwargs(settings.GITHUB_MANUAL_CONTRIBUTION_ASSESSMENT_PROMPT_NAME),
```
Comment (dmugtasimov) [inline thenewboston/contributions/models/contribution.py:76]: This helps development and debug where a developer can use a prompt with a particular tag by overriding a value in `settings.dev.py`

File: thenewboston/discord/bot.py [hunk 3]
```diff
@@ -20,6 +20,49 @@
 
 bot = commands.Bot('/', intents=intents)
 
+# TODO(dmu) HIGH: Cover bot logic with unittests: it is already complex enough
+
+
+def is_ia(author):
+    return author.id == settings.IA_DISCORD_USER_ID
```
Comment (dmugtasimov) [inline thenewboston/discord/bot.py:27]: This fixes the issue where anyone impersonate for IA just by having `_ai` in their username. It also allows not to force developers to have a bot with `_ai` string in their name

File: thenewboston/discord/bot.py [hunk 4]
```diff
@@ -20,6 +20,49 @@
...
+
+def messages_to_structured(messages):
+    structured_messages = []
+
+    prev_role = None
+    for message in messages:
+        content = message.content
+
+        if (role := map_author_structured(message.author)) == prev_role:
+            # We need to merge messages to prevent the following error from Anthropic
```
Comment (dmugtasimov) [inline thenewboston/discord/bot.py:50]: Anthropic has its limitations. We may need similar thing in the other place of IA interaction if we decide to use Anthropic there

File: thenewboston/discord/bot.py [hunk 5]
```diff
@@ -32,28 +75,25 @@ async def on_message_implementation(message):
         await message.reply('Please, register at https://thenewboston.com')
         return
 
-    # TODO(dmu) MEDIUM: Request message history just once and convert it to necessary format before LLM call
-    plain_text_message_history = await get_plain_text_message_history(message.channel)
+    messages = (await get_historical_messages(message.channel))[::-1]
```
Comment (dmugtasimov) [inline thenewboston/discord/bot.py:78]: Now we query historical messages just once

File: thenewboston/discord/bot.py [hunk 6]
```diff
@@ -20,6 +20,49 @@
 
 bot = commands.Bot('/', intents=intents)
 
+# TODO(dmu) HIGH: Cover bot logic with unittests: it is already complex enough
```
Comment (dmugtasimov) [inline thenewboston/discord/bot.py:23]: Saving time on not having unittests is overestimated

File: thenewboston/general/management/commands/llm.py [hunk 7]
```diff
@@ -2,22 +2,25 @@
 
 from django.conf import settings
 
-from thenewboston.general.clients.openai import OpenAIClient
+from thenewboston.general.clients.llm import LLMClient
 from thenewboston.general.commands import CustomCommand
+from thenewboston.general.utils.json import ResponseEncoder
 
 
 class Command(CustomCommand):
```
Comment (dmugtasimov) [inline thenewboston/general/management/commands/llm.py:10]: Was very handy for developing and debugging Anthropic integration

**Review by buckyroberts — APPROVED** [review]

**Pull request merged**